### DISCUSSION TITLE
NextJs 13: Is it possible to have a client components as a server component child?

### DISCUSSION THREAD

**Discussion by tiavina-mika** [vercel/next.js#45982] [Help]
Is it possible to have a client components as a server component child?

`app/clientComponent.js`
```
'use client'

const ClientChild = ({ children }) => {
  const [count, setCount] = useState();

  return (
    <div>
      <button onClick={() => setCount(count + 1)}>Count</button>
      <div>
        {children}
      </div>
    </div>
  )
}
```

`app/serverComponent.js`
```
const ServerParent = () => {
  return (
    <div>
      {products.map((product, index) => (
        <ClientChild key={index}>
          <h5>{product.name}</h5>
        </ClientChild>
      ))}
    </div>
  )
}
```

**Comment by icyJoseph** [comment]
Hi,

Yes, this is the fundamental idea behind server components, fetch and do as much as possible in the server, and at the ends of the React tree, so called leaves, put client components, which use interaction and what not.

There's but one caveat, the Props passed from Server Component to Client Component have to be serializable.

- https://beta.nextjs.org/docs/rendering/server-and-client-components#passing-props-from-server-to-client-components-serialization

It is worth noting that the inverse, Server as child of Client, is possible, but not that straight forward.

**Comment by denisso** [comment]
I will back up the words of [icyJoseph](https://github.com/icyJoseph) with my code.

[deploy](https://xewsnf-3000.preview.csb.app/), [code repository](https://github.com/denisso/next13-app-exp/tree/loading)

file structure 

```
app
│   globals.css
│   layout.module.scss
│   layout.tsx
│   loading.tsx
│   page.tsx
│
└───[page]
        page.tsx
```
file: app\[page]\page.tsx
```tsx
import { ServerComponent } from "@/components/ServerComponent";
import { ClientComponent } from "@/components/ClientComponent";

export default async function Page({ params }: { params: { page: string } }) {
  return (
    <ClientComponent header={params.page}>
      {/* @ts-expect-error Server Component */}
      <ServerComponent page={params.page} />
    </ClientComponent>
  );
}
export const dynamic = "force-dynamic";
```
about [@ts-expect-error Server Component](https://beta.nextjs.org/docs/data-fetching/fetching#asyncawait-in-server-components)
